test/WeakPartialOrderingTest.cpp

```diff
@@ -0,0 +1,853 @@
+#include "WeakPartialOrdering.h"
+
+#include <gtest/gtest.h>
+#include <set>
+#include <sstream>
+#include <sstream>
+#include <stack>
+#include <string>
+#include <unordered_map>
+
+using namespace sparta;
+
+using WpoIdx = uint32_t;
+
+class SimpleGraph2 final {
```

Comment:
Could you factor `SimpleGraph` from the WeakTopologicalOrderingTest into a separate file and reuse it here?